imgui_impl_vitagl.cpp

```diff
@@ -375,25 +380,31 @@ void ImGui_ImplVitaGL_NewFrame()
 
 	// Touch for mouse emulation
 	if (touch_usage){
-		SceTouchData touch;
-		sceTouchPeek(SCE_TOUCH_PORT_FRONT, &touch, 1);
-		if (touch.reportNum > 0){
-			mx = lerp(touch.report[0].x, 1920, 960);
-			my = lerp(touch.report[0].y, 1088, 544);
-		}
+		imgui_ImplVitaGL_PollTouch(&mx, &my, g_MousePressed);
 	}
 	
 	// Keys for mouse emulation
 	if (keys_usage){
 		SceCtrlData pad;
 		sceCtrlPeekBufferPositive(0, &pad, 1);
-		int lx = pad.lx - 127;
-		int ly = pad.ly - 127;
-		IN_RescaleAnalog(&lx, &ly, 30);
-		mx += lx >> 2;
-		my += ly >> 2;
-		g_MousePressed[0] = pad.buttons & SCE_CTRL_LTRIGGER;
-		g_MousePressed[1] = pad.buttons & SCE_CTRL_RTRIGGER;
+		int lx = (pad.lx - 127) * 256;
+		int ly = (pad.ly - 127) * 256;
+		IN_RescaleAnalog(&lx, &ly, 7680);
+		hires_x += lx;
+		hires_y += ly;
+		if (hires_x != 0 || hires_y != 0) {
+			// slow down pointer, could be made user-adjustable
+			int slowdown = 2048;
+			mx += hires_x / slowdown;
+			my += hires_y / slowdown;
+			hires_x %= slowdown;
+			hires_y %= slowdown;
+		}
+		if ((pad.buttons & SCE_CTRL_LTRIGGER) != (g_OldPad.buttons & SCE_CTRL_LTRIGGER))
```

Comment:
No it still works, because I also removed the reset of g_MouseButton to false. This is the correct way to do it and it is necessary to allow drag and drop and long clicks with both touch and keys, and, maybe, in the future also Bluetooth mice.